### PR TITLE
Fix session_name(): Cannot change session name when session is active

### PR DESCRIPTION
**Pull request by bscheshirwork**
https://github.com/php/php-src/commit/7f196e3

| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | [yii2-app-advanced #326](https://github.com/yiisoft/yii2-app-advanced/issues/326)

### CODE REVIEW

**Comment by bscheshirwork** [comment]
I got BC broken in child 

`yiisoft\yii2-redis\Session`

**Comment by bscheshirwork** [comment]
I can understand what the reason configure session name throw DI...

**Comment by bscheshirwork** [comment]
I will add check on "standard storage". 
Disabled for other. 

What about add this check for all engine-dependencies functions of `web\Sessions`?

**Comment by bscheshirwork** [comment]
related #15523

**Comment by samdark** [comment]
Should be fixed like https://github.com/yiisoft/yii2/blob/master/framework/web/Session.php#L421

**Comment by bscheshirwork** [comment]
Question: the unfrozen name is correct?

File: framework/web/Session.php [hunk 1]
```diff
@@ -322,7 +322,12 @@ public function getName()
      */
     public function setName($value)
     {
+        if ($this->getUseCustomStorage()) {
```
Comment (samdark) [inline framework/web/Session.php]: I don't think there's a need for it.

File: tests/framework/web/session/SessionTest.php [hunk 2]
```diff
@@ -70,4 +70,22 @@ public function testParamsAfterSessionStart()
...
+     * Test set name. Also check set name twice and after open
+     */
+    public function testSetName()
+    {
+        $session = new Session();
+        $session->setName('oldName');
+
+        $this->assertEquals('oldName', $session->getName());
+
+        $session->open();
```
Comment (samdark) [inline tests/framework/web/session/SessionTest.php]: Why do we need to call `open` here?

Comment (bscheshirwork) [reply tests/framework/web/session/SessionTest.php]: > check set name ... after open

Comment (samdark) [reply tests/framework/web/session/SessionTest.php]: Right.

File: framework/web/Session.php (hunk 1 above)
Comment (samdark) [reply framework/web/Session.php]: So why it's here?

**Comment by samdark** [comment]
Merged. Thanks!

**Comment by bscheshirwork** [comment]
> Removed unneeded code [skip ci]

```diff
-        if ($this->getUseCustomStorage()) {
-            return;
-        }
```
@samdark 
without this "unneeded code" the nested components like [redis session](https://github.com/yiisoft/yii2-redis/blob/master/src/Session.php) can use standard 

https://github.com/yiisoft/yii2/blob/86664eb50336f71e8dfcdf3871a33197cbbaa659/framework/web/Session.php#L945

this is (may be) BC broken

**Comment by bscheshirwork** [comment]
for example: we change storage to the `redis` in the config file. We can't change name from config file. We got a error.

**Comment by samdark** [comment]
Don't we need the same check then for setId, setSavePath, regenerateID, setName, setUseCookies etc.?

**Comment by bscheshirwork** [comment]
IMHO yes. For any non-redefine functional in children. Now. In future.
If in user code (in surce or in config) this functional will called = got a error.

But may be this is responsibility of a end user?

**Comment by samdark** [comment]
Yes. That could be done but in a separate pull request.

**Pull request merged**